.github/workflows/lfric_test.yml

```diff
@@ -75,6 +75,58 @@ jobs:
         pip install .[test]
         pip install jinja2
 
+    # PSyclone, compile and run MetOffice gungho_model on GPU
```

Comment:
I agree with the problem (it felt bad duplicating that much code for a 1-line difference) but github matrices operates on "jobs" not "steps" so I am not sure how to apply them here: it will probably involve creating separate jobs for each current step and a set-up job, with the "needs" clause to all jobs to the set-up. And will appear is different actions in the list here and the mirror.

Is that what you mean, or do you see an easier solution?

Reply:
Oh yes, I see what you mean: our current LFRic battery is a single job with each of the builds being a different step in that job. As you say, it looks like we'd need each step to be a job. That needs more thought so I'm happy to punt it for now.

Reply:
I've created #2740 for this.